### ISSUE TITLE
Whitelist not working for layots ?

### ISSUE THREAD

**Issue by quddoss** [maizzle/framework#956]
- Maizzle Version: v4.4.2
- Node.js Version: v14.21.2

I need to make a background gradient like css and like an image in the same classname (for different email clients ), but in the final result, the styles are inserted in the layout but the class is removed. Components keep both, styles and class, for me it is necessary that the class is saved and the styles are set ...
```js
// config.js
module.exports =  {
     removeUnusedCSS: {
         enabled: true,
         uglify: true,
         whitelist: ['.bg-primary-gradient']
     }
 }
```

```html
<style>
    .bg-primary-gradient {
        background-image: linear-gradient(...)
    }
</style>

// outlook hack
<style data-embed>
    .bg-primary-gradient\0{
        background-image: url("bg-primary-gradient.png")
        ....
    }
</style>
```

Input: 
```html 
// layouts/main.html
 ...
  <table class="w-full">
      <tr>
          <td class="bg-primary-gradient"> ... </td>
      </tr>
  </table>
...
```

Output:
```html
<table width="100%" style="width: 100%">
    <tr>
        <td style="background-image: linear-gradient(...)"></td>
    </tr>
</table>
```

Component output: (class remaining)
```html
<table width="100%" style="width: 100%">
    <tr>
        <td class="bg-gradient-primary" style="background-image: linear-gradient(...)"></td>
    </tr>
</table>
```

**Comment by cossssmin** [comment]
Unfortunately you can't really do that right now, enabling `removeUnusedCSS` will remove any possibly inlined CSS selectors in Layouts or Templates.

One thing to note is that you could create a Tailwind variant maybe like this:

```js
const plugin = require('tailwindcss/plugin')

module.exports = {
  plugins: [
    // ...
    plugin(function ({addVariant}) {
      addVariant('outlook', '&\\0')
    }),
  ],
}
```

And you could then use `outlook:bg-primary-gradient` to output `.bg-primary-gradient\0 {...}`.

However that doesn't help in this case, because Gmail will discard the entire `<style>` block where it finds a `\` character.

What is actually needed is a way to ensure safelisted selectors in your `removeUnusedCSS.whitelist` will be preserved even though they have been inlined. I'll try to come up with something.

**Comment by quddoss** [comment]
@cossssmin Okay, Thanks for plugin hint!

> However that doesn't help in this case, because Gmail will discard the entire <style> block where it finds a \ character.

Yeah, I know about it, that's why I put it in a separate style tag, and created special with image background, because outlook and yahoo doesn't support `linear-gradient()` ...

Have a nice day !

**Comment by cossssmin** [comment]
Released `4.4.4`, works fine with something like this now:

```xml
<push name="head">
  <style>
    .bg-primary-gradient\0 {
      background-image: url("bg-primary-gradient.png");
    }
  </style>
</push>

<x-main>
  <fill:template>
    <div class="bg-primary-gradient">...</div>
  </fill:template>
</x-main>
```
